src/pyodide/BUILD.bazel

```diff
@@ -207,25 +209,25 @@ data = wd_ts_bundle_capnp(
         "generated/pyodide-bucket.json",
     ],
     internal_modules = [
-        "generated/emscriptenSetup.js",
+        # "generated/emscriptenSetup.js",
     ] + glob(
         [
             "internal/*.ts",
             "internal/topLevelEntropy/*.ts",
             # The pool directory is only needed by typescript, it shouldn't be used at runtime.
-            "internal/pool/*.ts",
+            # "internal/pool/*.ts",
```

Comment:
We should run `tsc --noEmit` on this directory to make sure that our types are correct. Probably alongside the esbuild command, since esbuild by itself ignores types.

Reply:
Is there an easy way to configure esbuild to type check perhaps? I’d prefer not writing a whole new babel target for this

Reply:
There are a couple of esbuild plugins that could possibly be used or we could make our own. It isn't that common for people to do it this way though, usually the build script for an esbuild project looks like `tsc --noEmit && node esbuild.config.mjs`

Reply:
Well this is a lot more involved than you make it seem, I'm just restoring the pool directory here so it'll be type checked in the same way it was before this PR.